scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,183 @@
+ *       Trigger test nuts workflow for plugin with required parameters
+ */
+
+const manualRepoOverrides = {
+  'salesforce-alm': {
+    org: 'salesforce-cli',
+    repo: 'toolbelt',
+  },
+};
+
+const hasTestNuts = (module) => {
+  const results = execSync(`npm show ${module.name}@${module.version} scripts`);
+  return results.includes('test:nuts');
+};
+
+const getNpmForModule = (module, sections) => {
+  const results = execSync(`npm show ${module.name}@${module.version} ${['dist-tags', ...sections].join(' ')} --json`);
+  return JSON.parse(results);
+};
+
+const getOrgAndProject = (module) => {
+  const results = execSync(`npm show ${module.name} homepage`);
+  if (Reflect.ownKeys(manualRepoOverrides).includes(module.name)) {
+    return manualRepoOverrides[module.name];
+  }
+  const repoUrl = new URL(results.toString().trim());
+  const [, org, repo] = repoUrl.pathname.split('/');
+  return { org, repo };
+};
+
+const orgSlug = (org, repo) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+/**
+ * Triggers the just-nuts workflow for the named org/repo/branch
+ * @param sfdxVersion
+ * @param org
+ * @param repo
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsForProject = async (sfdxVersion, org, repo, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      sfdx_version: sfdxVersion,
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug(org, repo)}/pipeline`, { method: 'POST', json: body });
+};
+
+/**
+ * Triggers the nuts monitoring workflow
+ * @param jobData
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsMonitor = async (jobData, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      nuts_job_data: JSON.stringify(jobData),
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug('salesforcecli', 'sfdx-cli')}/pipeline`, {
+    method: 'POST',
+    json: body,
+  });
+};
+/**
+ * Makes a request to named URL. Ensures that auth token is applied.
+ * @param url
+ * @param options
+ * @returns {Promise<any>}
+ */
+const circle = async (url, options = {}) => {
+  const defaultOptions = {
+    method: 'GET',
+    headers: { 'Circle-Token': process.env.CIRCLECI_API_TOKEN, responseType: 'json', resolveBodyOnly: true },
+  };
+  options = { ...defaultOptions, ...options };
+  const response = await (async () => {
+    try {
+      return await got(url, options);
+    } catch (error) {
+      console.log(error);
+      throw new Error(error && error.response?.body ? error.response.body : error);
+    }
+  })();
+  return JSON.parse(response.body);
+};
+
+/**
+ * Examines a lit of plugins to determine which have NUT tests
+ * @param plugins
+ * @returns {*}
+ */
+const qualifyPluginsWithNonUnitTests = (modules) => {
```

Comment:
Reworked this function to take into account the version of a module when fetching details from nom,

`npm show sfdx-cli@latest-rc....` 

Npm is nice enough to returned the requested details related to the named version in the show command.